src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.Windows.cs

```diff
@@ -21,18 +19,32 @@ private TarEntry ConstructEntryForWriting(string fullPath, string entryName, Fil
 
             FileAttributes attributes = File.GetAttributes(fullPath);
 
+            bool isDirectory = (attributes & FileAttributes.Directory) != 0;
+
+            Interop.Kernel32.WIN32_FIND_DATA data = default;
+            Interop.Kernel32.GetFindData(fullPath, isDirectory, ignoreAccessDenied: false, ref data);
+
             TarEntryType entryType;
             if ((attributes & FileAttributes.ReparsePoint) != 0)
             {
-                entryType = TarEntryType.SymbolicLink;
+                if (data.dwReserved0 is Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_SYMLINK or
+                                        Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_MOUNT_POINT)
```

Comment:
This looks like another data point in favor of having an API for ReparseTags https://github.com/dotnet/runtime/issues/1908.